bootstrap/autoload.php

```diff
@@ -1,23 +1,26 @@
 <?php
-
 define('LARAVEL_START', microtime(true));
 
 /*
-|--------------------------------------------------------------------------
-| Register The Composer Auto Loader
-|--------------------------------------------------------------------------
-|
-| Composer provides a convenient, automatically generated class loader
-| for our application. We just need to utilize it! We'll require it
-| into the script here so we do not have to manually load any of
-| our application's PHP classes. It just feels great to relax.
-|
-*/
+  |--------------------------------------------------------------------------
+  | Register The Composer Auto Loader
+  |--------------------------------------------------------------------------
+  |
+  | Composer provides a convenient, automatically generated class loader
+  | for our application. We just need to utilize it! We'll require it
+  | into the script here so we do not have to manually load any of
+  | our application's PHP classes. It just feels great to relax.
+  |
+ */
+
+$plugin_dir = dirname(__DIR__)."/html/plugins";
```

Comment:
Don't edit this file: See https://getcomposer.org/doc/04-schema.md#psr-4

Reply:
@daryl-peterson please read the link I posted.  It does the same thing but is much nicer.  Thanks.

Reply:
Correct looking at the composer.json
"autoload": {
        "psr-4": {
            "App\\": "app",
            "LibreNMS\\": "LibreNMS",
            "LibreNMS\\Tests\\": "tests"
	}
}
Search path is listed for print_r($loader); Only relevant part here
[prefixDirsPsr4:Composer\Autoload\ClassLoader:private]
[LibreNMS\Tests\] => Array
(
    [0] => /local-librenms.org/vendor/composer/../../tests
)

[LibreNMS\] => Array
(
    [0] => /local-librenms.org/vendor/composer/../../LibreNMS
)

[App\] => Array
(
   [0] => /local-librenms.org/vendor/composer/../../app
   [1] => /local-librenms.org/vendor/composer/../laravel/laravel/app
)

Close but means all plugins would have to be int the same namespace as
LibreNMS\Plugins simply because the Plugins file does an include to the plugin file.
[classMap:Composer\Autoload\ClassLoader:private] => Array
(
	[LibreNMS\Plugins] => /local-librenms.org/vendor/composer/../../LibreNMS/Plugins.php
)

Even with the Plugins.php class doing an include a file like

html/plugins/DevReport/DevReport.php

LibreNMS\Plugins\DevReport
class DevReport
{

}
It will not work. because it expects the namespace to be LibreNMS\Plugins
LibreNMS\Plugins\DevReport will not work

So in order to be able to auto load a plugin without doing a manual include of the file is to either
add the 
"LibreNMS\\Plugins\\": "plugins" to the composer.json or add the path using the addPsr4 method to the loader.

Reply:
actually you would have the add 
"LibreNMS\Plugins\": "html/plugins"  to the psr-4 section of composer.json

Reply:
Exactly, that is much preferable imho.

Reply:
Great so I'll add "LibreNMS\Plugins"  to the composer.json and revert the autoload. What about the changes to Plugins.php are those good ?

Reply:
Yeah those seem fine to me.  Not a fan of long nested ifs myself.

But those changes will require a little testing.  Testing time is short on the maintainers side of things, so you might have to wait awhile before it is merged.

Reply:
No problem. Thanks for the input